prow/proxy-common.inc

```diff
@@ -47,9 +48,10 @@ if [[ -n "${GOOGLE_APPLICATION_CREDENTIALS:-}" ]]; then
 fi
 
 # Override envoy.
-if [[ "${ENVOY_REPOSITORY:-}" && "${ENVOY_SHA:-}" && "${ENVOY_PREFIX:-}" ]]; then
+if [[ "${ENVOY_REPOSITORY:-}" && "${ENVOY_PREFIX:-}" ]]; then
   TMP_DIR=$(mktemp -d -t envoy-XXXXXXXXXX)
   trap 'rm -rf ${TMP_DIR:?}' EXIT
+  ENVOY_SHA="${ENVOY_SHA:-$(grep -Pom1 "^ENVOY_SHA = \"\K[a-zA-Z0-9]+" "$WORKSPACE")}"
```

Comment:
not familiar with bazel at all - is there a native way to read the variable instead of parsing it out?

If not lgtm, thanks!

Reply:
I am not a strong bazeler either. I can research a bit more into this.

Reply:
It's better to use a well-structured bazel file that is shared. Something like https://github.com/envoyproxy/envoy/blob/master/bazel/repository_locations.bzl

Reply:
that one doesn't have git sha in structured way either (only as a part of URL). 

btw What's the purpose here? download same tar from different server?

Reply:
> It's better to use a well-structured bazel file that is shared. Something like...

@kyessenov a native bazel solution would be great. I tried that initially and failed to provide a robust solution #2496. I plan to iterate on this as I develop more bazel knowledge.

> btw What's the purpose here? download same tar from different server?

@lizan yes, the idea here is to be able to pull same SHA from different fork/mirror https://github.com/istio/test-infra/pull/2091#discussion_r345572359. More specifically, to swap out the `ENVOY_REPOSITORY` with a *private* fork while retaining the configured SHA.